From 721b87a4b202d19cb9d6dfb212b489b37bb8b1ff Mon Sep 17 00:00:00 2001
From: dEFIX <1132404+911gt3@users.noreply.github.com>
Date: Tue, 17 Aug 2021 13:50:02 +0800
Subject: [PATCH 4/4] arm64: dts: rockchip: Add support for R2C

The only difference between R2C and R2S is the Ethernet PHY chip.

Here are the key notes for dts:
1) rgmii's drive strength should be 8mA
2) reset-assert-us should be 15000
3) The fully compatible tx/rx delay is 0x22/0x12

To remove the phy-handle, the following patch is required.
 ("stmmac: Use phy_find_first() for invalid phy_addr")

Signed-off-by: dEFIX <1132404+911gt3@users.noreply.github.com>
---
 .../boot/dts/rockchip/rk3328-nanopi-r2s.dts   | 23 +++++++++++--------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts
index 5e4c08a92e86..108cc3a4fd0b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts
@@ -114,6 +114,7 @@
 		regulator-name = "vcc_rtl8153";
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
+		off-on-delay-us = <5000>;
 		enable-active-high;
 	};
 
@@ -208,14 +209,16 @@
 	assigned-clocks = <&cru SCLK_MAC2IO>, <&cru SCLK_MAC2IO_EXT>;
 	assigned-clock-parents = <&gmac_clk>, <&gmac_clk>;
 	clock_in_out = "input";
-	phy-handle = <&rtl8211e>;
 	phy-mode = "rgmii";
 	phy-supply = <&vcc_io_33>;
 	pinctrl-0 = <&rgmiim1_pins>;
 	pinctrl-names = "default";
-	rx_delay = <0x18>;
+	rx_delay = <0x12>;
 	snps,aal;
-	tx_delay = <0x24>;
+	snps,reset-active-low;
+	snps,reset-delays-us = <0 15000 50000>;
+	snps,reset-gpio = <&gpio1 RK_PC2 GPIO_ACTIVE_LOW>;
+	tx_delay = <0x22>;
 	status = "okay";
 
 	mdio {
@@ -224,14 +227,14 @@
 		#size-cells = <0>;
 
 		rtl8211e: ethernet-phy@1 {
-			compatible = "ethernet-phy-id001c.c915",
-				     "ethernet-phy-ieee802.3-c22";
+			compatible = "ethernet-phy-ieee802.3-c22";
 			reg = <1>;
-			pinctrl-0 = <&eth_phy_reset_pin>;
-			pinctrl-names = "default";
-			reset-assert-us = <10000>;
-			reset-deassert-us = <50000>;
-			reset-gpios = <&gpio1 RK_PC2 GPIO_ACTIVE_LOW>;
+		};
+
+		ethphy3: ethernet-phy@3 {
+			reg = <3>;
+			interrupt-parent = <&gpio2>;
+			interrupts = <RK_PC4 IRQ_TYPE_EDGE_FALLING>;
 		};
 	};
 };
-- 
2.25.1

